pkg/protocol/sybilprotection/sybilprotectionv1/performance/performance.go

```diff
@@ -65,26 +71,69 @@ func (t *Tracker) TrackValidationBlock(block *blocks.Block) {
 
 	t.performanceFactorsMutex.Lock()
 	defer t.performanceFactorsMutex.Unlock()
+	isCommitteeMember, err := t.isCommitteeMember(block.ID().Index(), block.ProtocolBlock().IssuerID)
+	if err != nil {
+		t.errHandler(ierrors.Errorf("failed to check if account %s is committee member", block.ProtocolBlock().IssuerID))
+
+		return
+	}
 
-	performanceFactors, err := t.performanceFactorsFunc(block.ID().Index())
+	if isCommitteeMember {
+		t.trackCommitteeMemberPerformance(validatorBlock, block)
+	}
+}
+
+func (t *Tracker) trackCommitteeMemberPerformance(validationBlock *iotago.ValidationBlock, block *blocks.Block) {
+	validatorPerformances, err := t.validatorPerformancesFunc(block.ID().Index())
 	if err != nil {
 		t.errHandler(ierrors.Errorf("failed to load performance factor for slot %s", block.ID().Index()))
 		return
 	}
-	pf, err := performanceFactors.Load(block.ProtocolBlock().IssuerID)
+	validatorPerformance, err := validatorPerformances.Load(block.ProtocolBlock().IssuerID)
 	if err != nil {
 		t.errHandler(ierrors.Errorf("failed to load performance factor for account %s", block.ProtocolBlock().IssuerID))
 	}
-
-	// TODO: store highest supported version per validator?
-	_ = validatorBlock.HighestSupportedVersion
-
-	err = performanceFactors.Store(block.ProtocolBlock().IssuerID, pf+1)
+	// key not found
+	if validatorPerformance == nil {
+		validatorPerformance = model.NewValidatorPerformance()
+	}
+	updatedPerformance := t.updateSlotPerformanceBitMap(validatorPerformance, block.ID().Index(), block.ProtocolBlock().IssuingTime)
+	if updatedPerformance.BlockIssuedCount == t.apiProvider.APIForSlot(block.ID().Index()).ProtocolParameters().RewardsParameters().ValidatorBlocksPerSlot {
+		// no need to store larger number and we can fit into uint8
+		updatedPerformance.BlockIssuedCount = t.apiProvider.APIForSlot(block.ID().Index()).ProtocolParameters().RewardsParameters().ValidatorBlocksPerSlot + 1
+	} else {
+		updatedPerformance.BlockIssuedCount++
+	}
+	updatedPerformance.HighestSupportedVersionAndHash = model.VersionAndHash{
+		Version: validationBlock.HighestSupportedVersion,
+		Hash:    validationBlock.ProtocolParametersHash,
+	}
+	err = validatorPerformances.Store(block.ProtocolBlock().IssuerID, updatedPerformance)
 	if err != nil {
 		t.errHandler(ierrors.Errorf("failed to store performance factor for account %s", block.ProtocolBlock().IssuerID))
 	}
 }
 
+func (t *Tracker) updateSlotPerformanceBitMap(pf *model.ValidatorPerformance, slotIndex iotago.SlotIndex, issuingTime time.Time) *model.ValidatorPerformance {
+	if pf == nil {
+		return pf
+	}
+	subslotIndex := t.subslotIndex(slotIndex, issuingTime)
+	// set bit at subslotIndex to 1 to indicate activity in that subslot
+	pf.SlotActivityVector = pf.SlotActivityVector | (1 << subslotIndex)
```

Comment:
I don't know if this is intentional, but the last bit of SlotActivityVector is always 0, no? since subslotIndex is defined in line 134 starting at 1

Reply:
Yep, you are right! Started the indexing from 0 instead of 1